test/test_distributions.py

```diff
@@ -563,7 +577,65 @@ def test_gamma_sample_grad(self):
                                        'max error {}'.format(rel_error.max()),
                                        'at alpha={}, x={}'.format(alpha, x[rel_error.argmax()])]))
 
-    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    @unittest.skipIf(not TEST_NUMPY, "NumPy not found")
+    def test_pareto_shape(self):
+        scale = Variable(torch.randn(2, 3).abs(), requires_grad=True)
+        alpha = Variable(torch.randn(2, 3).abs(), requires_grad=True)
+        scale_1d = torch.randn(1).abs()
+        alpha_1d = torch.randn(1).abs()
+        self.assertEqual(Pareto(scale, alpha).sample().size(), (2, 3))
+        self.assertEqual(Pareto(scale, alpha).sample_n(5).size(), (5, 2, 3))
+        self.assertEqual(Pareto(scale_1d, alpha_1d).sample_n(1).size(), (1, 1))
+        self.assertEqual(Pareto(scale_1d, alpha_1d).sample().size(), (1,))
+        self.assertEqual(Pareto(1.0, 1.0).sample().size(), (1,))
+        self.assertEqual(Pareto(1.0, 1.0).sample_n(1).size(), (1,))
+
+        def ref_log_prob(idx, x, log_prob):
+            s = scale.data.view(-1)[idx]
+            a = alpha.data.view(-1)[idx]
+            expected = scipy.stats.pareto.logpdf(x, a, scale=s)
+            self.assertAlmostEqual(log_prob, expected, places=3)
+
+        self._check_log_prob(Pareto(scale, alpha), ref_log_prob)
+
+    @unittest.skipIf(not TEST_NUMPY, "NumPy not found")
+    def test_pareto_sample(self):
+        set_rng_seed(0)  # see Note [Randomized statistical tests]
+        for scale, alpha in product([0.1, 1.0, 5.0], [0.1, 1.0, 10.0]):
+            self._check_sampler_sampler(Pareto(scale, alpha),
+                                        scipy.stats.pareto(alpha, scale=scale),
+                                        'Pareto(scale={}, alpha={})'.format(scale, alpha))
+
+    @unittest.skipIf(not TEST_NUMPY, "NumPy not found")
+    def test_pareto_sample_grad(self):
```

Comment:
This test is unnecessary. It is really only needed for `Gamma` and `Dirichlet` because they implement custom reparameterized gradients `standard_gamma_grad()` and `dirichlet_grad()`.